x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/sample_logs_input.tsx

```diff
@@ -12,46 +12,105 @@ import { isPlainObject } from 'lodash/fp';
+ */
+export const parseJSONArray = (
+  fileContent: string
+): { entries: unknown[]; pathToEntries: string[]; errorNoArrayFound: boolean } => {
+  const jsonContent = JSON.parse(fileContent);
+  if (Array.isArray(jsonContent)) {
+    return { entries: jsonContent, pathToEntries: [], errorNoArrayFound: false };
+  }
+  if (typeof jsonContent === 'object' && jsonContent !== null) {
+    const arrayKeys = Object.keys(jsonContent).filter((key) => Array.isArray(jsonContent[key]));
+    if (arrayKeys.length === 1) {
+      const key = arrayKeys[0];
+      return {
+        entries: jsonContent[key],
+        pathToEntries: [key],
+        errorNoArrayFound: false,
+      };
+    }
+  }
+  return { errorNoArrayFound: true, entries: [], pathToEntries: [] };
+};
+
 /**
  * Parse the logs sample file content (json or ndjson) and return the parsed logs sample
  */
 const parseLogsContent = (
   fileContent: string | undefined
-): { error?: string; isTruncated?: boolean; logsSampleParsed?: string[] } => {
+): {
+  error?: string;
+  isTruncated?: boolean;
+  logsSampleParsed?: string[];
+  sampleFormat?: SampleFormat;
+} => {
   if (fileContent == null) {
     return { error: i18n.LOGS_SAMPLE_ERROR.CAN_NOT_READ };
   }
-  let parsedContent;
+  let parsedContent: unknown[];
+  let sampleFormat: SampleFormat;
+
   try {
-    parsedContent = fileContent
-      .split('\n')
-      .filter((line) => line.trim() !== '')
-      .map((line) => JSON.parse(line));
+    parsedContent = parseNDJSON(fileContent);
 
     // Special case for files that can be parsed as both JSON and NDJSON:
-    //   for a one-line array [] -> extract its contents
-    //   for a one-line object {} -> do nothing
-    if (
-      Array.isArray(parsedContent) &&
-      parsedContent.length === 1 &&
-      Array.isArray(parsedContent[0])
-    ) {
+    //   for a one-line array [] -> extract its contents (it's a JSON)
+    //   for a one-line object {} -> do nothing (keep as NDJSON)
+    if (parsedContent.length === 1 && Array.isArray(parsedContent[0])) {
       parsedContent = parsedContent[0];
+      sampleFormat = { name: 'json', json_path: [] };
+    } else {
+      sampleFormat = { name: 'ndjson', multiline: false };
     }
   } catch (parseNDJSONError) {
     try {
-      parsedContent = JSON.parse(fileContent);
+      const { entries, pathToEntries, errorNoArrayFound } = parseJSONArray(fileContent);
+      if (errorNoArrayFound) {
+        return { error: i18n.LOGS_SAMPLE_ERROR.NOT_ARRAY };
+      }
+      parsedContent = entries;
+      sampleFormat = { name: 'json', json_path: pathToEntries };
     } catch (parseJSONError) {
-      return { error: i18n.LOGS_SAMPLE_ERROR.CAN_NOT_PARSE };
+      try {
+        parsedContent = parseNDJSON(fileContent, true);
+        sampleFormat = { name: 'ndjson', multiline: true };
+      } catch (parseMultilineNDJSONError) {
+        return { error: i18n.LOGS_SAMPLE_ERROR.CAN_NOT_PARSE };
+      }
```

Comment:
Where does this scenario fall: [Sample logs](https://github.com/elastic/integrations/blob/main/packages/crowdstrike/data_stream/falcon/_dev/deploy/docker/sample_logs/falcon-sample.log)

`json multiline 'with_path'` Or `ndjson multiline 'with_path'`

```json
{
    "metadata": {
        "eventType": "FirewallMatchEvent"
    },
    "event": {
        "Ipv": "ipv4",
        "HostName": "TESTDEVICE01",
        "Timestamp": "2020-07-20T12:41:44Z"
    }
}
{
    "metadata": {
        "eventType": "IncidentSummaryEvent"
    },
    "event": {
        "IncidentStartTime": 1595005316,
        "IncidentEndTime": 1595005316
    }
}
{
    "metadata": {
        "eventType": "AuthActivityAuditEvent"
    },
    "event": {
        "UserId": "first.last@company.com",
        "UTCTimestamp": 1595247970
    }
}
```

Reply:
It's `ndjson` multiline.

Reply:
OK. Sounds like the scenario is in control.